common/v2/features/AddAccount/components/DeterministicWallets.tsx

```diff
@@ -230,13 +234,13 @@ class DeterministicWalletsClass extends React.PureComponent<Props, State> {
     );
   }
 
-  private renderWalletRow(wallet: deterministicWalletsTypes.DeterministicWalletData) {
-    const { network, addressLabels } = this.props;
+  private renderWalletRow(wallet: deterministicWalletsTypes.DeterministicWalletData, network : Network | StaticNetworkConfig | CustomNetworkConfig) {
+    const { addressLabels } = this.props;
     const { selectedAddress } = this.state;
     const label = addressLabels[wallet.address.toLowerCase()];
 
     let blockExplorer;
-    if (!network.isCustom) {
+    if (!network.isCustom && network.blockExplorer) {
       blockExplorer = network.blockExplorer;
     } else {
       blockExplorer = {
```

Comment:
Ouch!
This shouldn't be done inside the component.
Prefer updating `makeExplorer` with the default value `https://ethplorer.io/address/...`

Reply:
I'm unsure of what you mean here as it also depends on how you want me to access network details.

Reply:
To access networks please use:
```
import { NetworkContext } from 'v2/services/Store';
...
const { networks } = useContext(NetworkContext
```

In which case the `blockExplorer` should be a valid object.

Reply:
I messaged you on Slack with some possiblities on how to do this. The problem is that the existing components are defined as classes and not functions, thus using context isn't as easy as your example.